test/e2e/framework/helpers/yunikorn/rest_api_utils.go

```diff
@@ -419,30 +419,31 @@ func GetFailedHealthChecks() (string, error) {
 	return failCheck, nil
 }
 
-func (c *RClient) GetQueue(partition string, queueName string) (*dao.PartitionQueueDAOInfo, error) {
-	queues, err := c.GetQueues(partition)
+func (c *RClient) GetQueue(partition string, queueName string, withChildren bool) (*dao.PartitionQueueDAOInfo, error) {
+	req, err := c.newRequest("GET", fmt.Sprintf(configmanager.QueuePath, partition, queueName), nil)
 	if err != nil {
 		return nil, err
 	}
-	if queueName == "root" {
-		return queues, nil
+	if withChildren {
+		q := req.URL.Query()
+		q.Add("subtree", "true")
+		req.URL.RawQuery = q.Encode()
 	}
 
-	var allSubQueues = queues.Children
-	for _, subQ := range allSubQueues {
-		if subQ.QueueName == queueName {
-			return &subQ, nil
-		}
+	var queue *dao.PartitionQueueDAOInfo
+	_, err = c.do(req, &queue)
+	if queue == nil {
+		return nil, fmt.Errorf("QueueInfo not found: %s", queueName)
 	}
-	return nil, fmt.Errorf("QueueInfo not found: %s", queueName)
+	return queue, err
```

Comment:
```suggestion
	return queue, nil
```

Should we return `nil` if no error be observed?

Reply:
Thank you for your suggestion. The adjustment has been made.